src/main/java/io/github/cottonmc/energy/api/EnergyStorage.java

```diff
@@ -1,48 +1,20 @@
 package io.github.cottonmc.energy.api;
 
-import io.github.cottonmc.energy.event.PowerGenEvent;
-import io.github.cottonmc.energy.util.PacketTier;
-import net.fabricmc.fabric.util.HandlerArray;
-import net.minecraft.block.entity.BlockEntity;
-
-import java.util.function.Consumer;
-
-public interface EnergyStorage {
+public interface EnergyStorage extends EnergyProducer, EnergyConsumer {
 
     /**
-     * @return the maximum amount of energy your storage can hold.
+     * @return the maximum amount of energy the storage can hold.
      */
-    int getMaxCapacity();
+    int getMaxEnergy();
 
     /**
-     * @return the current amount of energy your storage is holding.
+     * @return the current amount of energy the storage is holding.
      */
     int getCurrentEnergy();
 
     /**
-     * @return what tier of energy packet your storage can accept.
+     * @return the highest tier of energy this device is capable of working with under any circumstances.
      */
     PacketTier getPacketTier();
 
-    /**
-     * @param amount how much energy to be generated/
-     * @return whether the machine has the capacity for generating that amount of energy.
-     */
-    default boolean canGenerateEnergy(int amount) {
-        return (getMaxCapacity() - getCurrentEnergy()) >= amount;
-    }
-
-    /**
-     * Generate an amount of power, without needing to worry about packet sizes.
-     * DO NOT OVERRIDE WITHOUT CALLING THE EVENT HANDLER. DOING SO CAN AND WILL BREAK THINGS.
-     *
-     * @param amount The amount of energy to generate.
-     */
-    default void generateEnergy(int amount) {
-        if (canGenerateEnergy(amount)) {
-            for (Consumer<BlockEntity> handler : ((HandlerArray<Consumer<BlockEntity>>) PowerGenEvent.GENERATE_POWER).getBackingArray()) {
```

Comment:
Has this event call been moved anywhere yet? I do want to keep it in just so mods don't have to go through a super-contrived process to hook into generation events for stuff like stat tracking or pollution.